service/src/main/java/com/ge/predix/acs/attribute/cache/AttributeCacheFactory.java

```diff
@@ -1,17 +1,37 @@
 package com.ge.predix.acs.attribute.cache;
 
+import org.springframework.data.redis.core.RedisTemplate;
+
 // CHECKSTYLE:OFF: FinalClass
 public class AttributeCacheFactory {
 
+    private static String SPRING_PROFILES = System.getenv("SPRING_PROFILES_ACTIVE");
+
     private AttributeCacheFactory() {
         throw new AssertionError();
     }
 
-    public static AttributeCache createResourceAttributeCache(final long maxStorageInMegabytes) {
+    public static AttributeCache createResourceAttributeCache(final long maxCachedIntervalMinutes,
```

Comment:
I suggest you have one factory for each cache type, which is enabled using @Profile annotation.